### PR TITLE
Let bitfields accept any ordinal, not just its individual flags.

### LINKED ISSUE #335: Bitfields are erroneously treated as enums
I may be holding this wrong, but using the latest gdext 5d6725f4 and upgrading to Godot 4.1, I've started seeing the following error message when attempting to call `add_surface_from_arrays` which appears to originate from within `ArrayMesh` in gdext.

```
thread '<unnamed>' panicked at 'ordinal 0 does not map to any enumerator', C:\Users\Sean\.cargo\git\checkouts\gdext-76630c89719e160c\5d6725f\godot-core\src\obj\traits.rs:138:32
```

### My Code

```rust
 let mut mesh = ArrayMesh::new();
 mesh.add_surface_from_arrays(PrimitiveType::PRIMITIVE_TRIANGLES, surface);
```

I have also tried the following to see if I could prevent gdext from using the default 0:

```rust
 let mut mesh = ArrayMesh::new();
 mesh.add_surface_from_arrays_ex(PrimitiveType::PRIMITIVE_TRIANGLES, surface)
     .flags(ArrayFormat::ARRAY_FORMAT_VERTEX)
     .done();
```

### Debugging

Following the stack trace (attached at the bottom of this issue), I did poke around in the generated code for ArrayMesh. Flags does appear to be initialised with a 0, which I suspect is the cause of our panic. I'm not really sure what else we could replace this with?

```rust
#[allow(clippy::wrong_self_convention, clippy::redundant_field_names, clippy::needless_update)]
impl < 'a > ExAddSurfaceFromArrays < 'a > {
fn new(surround_object: &'a mut re_export::ArrayMesh, primitive: crate::engine::mesh::PrimitiveType, arrays: VariantArray,) -> Self {
    Self {
        surround_object, primitive, arrays, blend_shapes: Array::new(), lods: Dictionary::new(), flags: crate::obj::EngineEnum::from_ord(0),
    }
}
```

### Stack Trace

<details>
<summary>Click to see stack full stack trace</summary>

If it helps, I can generate the stack trace with `RUST_BACKTRACE=full`, just let me know!

```
INFO.rs: Starting load_chunks_async...
thread '<unnamed>' panicked at 'ordinal 0 does not map to any enumerator', C:\Users\Sean\.cargo\git\checkouts\gdext-76630c89719e160c\5d6725f\godot-core\src\obj\traits.rs:138:32
stack backtrace:
   0: std::panicking::begin_panic_handler
             at /rustc/37998ab508d5d9fa0d465d7b535dc673087dda8f/library\std\src\panicking.rs:593
   1: core::panicking::panic_fmt
             at /rustc/37998ab508d5d9fa0d465d7b535dc673087dda8f/library\core\src\panicking.rs:67
   2: godot_core::obj::traits::EngineEnum::from_ord::closure$0<godot_core::gen::classes::mesh::ArrayFormat>
             at C:\Users\Sean\.cargo\git\checkouts\gdext-76630c89719e160c\5d6725f\godot-core\src\obj\traits.rs:138
   3: enum2$<core::option::Option<godot_core::gen::classes::mesh::ArrayFormat> >::unwrap_or_else<godot_core::gen::classes::mesh::ArrayFormat,godot_core::obj::traits::EngineEnum::from_ord::closure_env$0<godot_core::gen::classes::mesh::ArrayFormat> >
             at /rustc/37998ab508d5d9fa0d465d7b535dc673087dda8f\library\core\src\option.rs:979
   4: godot_core::obj::traits::EngineEnum::from_ord<godot_core::gen::classes::mesh::ArrayFormat>
             at C:\Users\Sean\.cargo\git\checkouts\gdext-76630c89719e160c\5d6725f\godot-core\src\obj\traits.rs:137
   5: godot_core::gen::classes::array_mesh::ExAddSurfaceFromArrays::new
             at C:\Users\Sean\.cargo\git\checkouts\gdext-76630c89719e160c\5d6725f\godot-core\src\gen\classes\array_mesh.rs:641
   6: godot_core::gen::classes::array_mesh::re_export::ArrayMesh::add_surface_from_arrays_ex
             at C:\Users\Sean\.cargo\git\checkouts\gdext-76630c89719e160c\5d6725f\godot-core\src\gen\classes\array_mesh.rs:270
   7: bridge_gd::model::segment::loader::load_vertex_buffer_async
             at A:\Workspace\bridge\bridge-gd\src\model\segment\loader.rs:119
   8: bridge_gd::model::segment::loader::load_segment_async
             at A:\Workspace\bridge\bridge-gd\src\model\segment\loader.rs:32
   9: bridge_gd::model::loader::load_model_async
             at A:\Workspace\bridge\bridge-gd\src\model\loader.rs:21
  10: bridge_gd::loader::load_chunk_async
             at A:\Workspace\bridge\bridge-gd\src\loader.rs:45
  11: core::ops::function::FnMut::call_mut<void (*)(bridge::chunk::Chunk),tuple$<bridge::chunk::Chunk> >
             at /rustc/37998ab508d5d9fa0d465d7b535dc673087dda8f\library\core\src\ops\function.rs:166
  12: core::iter::traits::iterator::Iterator::for_each::call::closure$0<bridge::chunk::Chunk,void (*)(bridge::chunk::Chunk)>
             at /rustc/37998ab508d5d9fa0d465d7b535dc673087dda8f\library\core\src\iter\traits\iterator.rs:853
  13: core::iter::traits::iterator::Iterator::fold<alloc::vec::into_iter::IntoIter<bridge::chunk::Chunk,alloc::alloc::Global>,tuple$<>,core::iter::traits::iterator::Iterator::for_each::call::closure_env$0<bridge::chunk::Chunk,void (*)(bridge::chunk::Chunk)> >
             at /rustc/37998ab508d5d9fa0d465d7b535dc673087dda8f\library\core\src\iter\traits\iterator.rs:2481
  14: core::iter::traits::iterator::Iterator::for_each<alloc::vec::into_iter::IntoIter<bridge::chunk::Chunk,alloc::alloc::Global>,void (*)(bridge::chunk::Chunk)>
             at /rustc/37998ab508d5d9fa0d465d7b535dc673087dda8f\library\core\src\iter\traits\iterator.rs:856
  15: bridge_gd::loader::load_chunk_tree_async
             at A:\Workspace\bridge\bridge-gd\src\loader.rs:35
  16: bridge_gd::singleton::BridgeSingleton::load_classic_lvl
             at A:\Workspace\bridge\bridge-gd\src\singleton.rs:69
  17: bridge_gd::level::node::LevelNode::generate
             at A:\Workspace\bridge\bridge-gd\src\level\node.rs:32
  18: bridge_gd::level::node::impl$12::ready
             at A:\Workspace\bridge\bridge-gd\src\level\node.rs:65
  19: bridge_gd::level::node::impl$15::__virtual_call::function::closure$0
             at A:\Workspace\bridge\bridge-gd\src\level\node.rs:48
  20: core::ops::function::FnOnce::call_once<bridge_gd::level::node::impl$15::__virtual_call::function::closure_env$0,tuple$<ptr_mut$<godot_ffi::gen::gdextension_interface::__GdextClassInstance>,tuple$<> > >
             at /rustc/37998ab508d5d9fa0d465d7b535dc673087dda8f\library\core\src\ops\function.rs:250
  21: godot_core::builtin::meta::signature::impl$1::ptrcall<tuple$<> >
             at C:\Users\Sean\.cargo\git\checkouts\gdext-76630c89719e160c\5d6725f\godot-core\src\builtin\meta\signature.rs:154
  22: bridge_gd::level::node::impl$15::__virtual_call::function
             at A:\Workspace\bridge\bridge-gd\src\level\node.rs:48
  23: <unknown>
  24: ZN6embree13TaskScheduler15removeSchedulerERKNS_3RefIS0_EE
  25: ZN6embree13TaskScheduler15removeSchedulerERKNS_3RefIS0_EE
  26: ZN6embree13TaskScheduler15removeSchedulerERKNS_3RefIS0_EE
  27: ZN6embree13TaskScheduler15removeSchedulerERKNS_3RefIS0_EE
  28: ZN6embree13TaskScheduler15removeSchedulerERKNS_3RefIS0_EE
  29: oidnExecuteFilter
  30: oidnExecuteFilter
  31: ZN6embree13TaskScheduler15removeSchedulerERKNS_3RefIS0_EE
  32: ZN6embree13TaskScheduler15removeSchedulerERKNS_3RefIS0_EE
  33: oidnExecuteFilter
  34: oidnExecuteFilter
  35: ZN6embree13TaskScheduler15removeSchedulerERKNS_3RefIS0_EE
  36: ZN6embree13TaskScheduler15removeSchedulerERKNS_3RefIS0_EE
  37: oidnExecuteFilter
  38: oidnExecuteFilter
  39: ZN6embree13TaskScheduler15removeSchedulerERKNS_3RefIS0_EE
  40: ZN6embree13TaskScheduler15removeSchedulerERKNS_3RefIS0_EE
  41: ZN6embree13TaskScheduler15removeSchedulerERKNS_3RefIS0_EE
  42: ZN6embree13TaskScheduler15removeSchedulerERKNS_3RefIS0_EE
  43: ZN6embree13TaskScheduler15removeSchedulerERKNS_3RefIS0_EE
  44: ZN6embree13TaskScheduler15removeSchedulerERKNS_3RefIS0_EE
  45: ZN6embree13TaskScheduler15removeSchedulerERKNS_3RefIS0_EE
  46: ZN6embree13TaskScheduler15removeSchedulerERKNS_3RefIS0_EE
  47: <unknown>
  48: <unknown>
  49: ZN6embree13TaskScheduler15removeSchedulerERKNS_3RefIS0_EE
  50: <unknown>
  51: <unknown>
  52: BaseThreadInitThunk
  53: RtlUserThreadStart
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```

</details>

### PR DESCRIPTION
**Pull request by lilizoey**
Should we try to do some verification to make sure that the bitfield only has its defined flags set? I'm leaning towards no personally.

closes #335

### CODE REVIEW

File: itest/rust/src/enum_test.rs [hunk 1]
```diff
@@ -6,7 +6,9 @@
 
 use crate::itest;
 use godot::engine::input::CursorShape;
-use godot::engine::time;
+use godot::engine::mesh::PrimitiveType;
+use godot::engine::{time, ArrayMesh};
+use godot::prelude::varray;
```
Comment (Bromeon) [inline itest/rust/src/enum_test.rs:11]: Maybe use the one from `builtin` here

**Comment by GodotRust** [comment]
API docs are being generated and will be shortly available at: https://godot-rust.github.io/docs/gdext/pr-342

**Review by Bromeon — COMMENTED** [review]
No, I think it's fine for now. We can always think about a more type-safe abstraction like [`bitflags`](lib.rs/bitflags) in the future.

**Comment by Bromeon** [comment]
Thank you!

**Pull request merged**